test/browser-tests/push-worker/get-notifications/get-notifications.js

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright 2016 Google Inc. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// This is a test and we want descriptions to be useful, if this
+// breaks the max-length, it's ok.
+
+/* eslint-disable max-len, no-unused-expressions */
+/* eslint-env browser, mocha */
+
+'use strict';
+
+describe('Test PushWorker.getNotifications()', function() {
+  const swUtils = window.goog.SWHelper;
+  const serviceWorkersFolder = '/test/browser-tests/push-worker/get-notifications/serviceworkers';
+
+  if (Notification.permission !== 'granted') {
```

Comment:
It seems like it makes sense to test the function even without permission. Have a test that calls `getNotifications` before any are created, just to check that it doesn't error (regardless of whether permission is granted) and returns a valid result (presumably empty array).

Put the permission check after that.